src/Symfony/Component/Security/Http/Authenticator/AccessToken/DefaultAuthenticationFailureHandler.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator\AccessToken;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Http\Authentication\AuthenticationFailureHandlerInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Provides an implementation of the RFC6750 of an authentication via
+ * an access token.
+ *
+ * @author Florent Morselli <florent.morselli@spomky-labs.com>
+ */
+class DefaultAuthenticationFailureHandler implements AuthenticationFailureHandlerInterface
+{
+    private ?TranslatorInterface $translator = null;
+
+    public function __construct(private readonly ?string $realm = null)
+    {
+    }
+
+    public function setTranslator(?TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    public function onAuthenticationFailure(Request $request, AuthenticationException $exception): Response
```

Comment:
I actually quite like how [the default failure handling is done in the json login authenticator](https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/Security/Http/Authenticator/JsonLoginAuthenticator.php#L111-L124), by making `failureHandler` nullable in the authenticator and moving this code to the `onAuthenticationFailure()` method of the authenticator.

This embraces the benefit of authenticators of having everything for the authentication part together in a single file.

Reply:
Not sure to get it. I decided to do it that way in order to be compliant with the response showed in the RFC6750 and [RFC9110](https://datatracker.ietf.org/doc/html/rfc9110#section-11.6.1) about the `WWW-Authenticate` header when no authentication failure is set.

Reply:
With the change proposed in https://github.com/symfony/symfony/pull/46428#discussion_r934710681, it was indeed necessary to allow the failure handler to be `null`.
I renamed the `DefaultAuthenticationFailureHandler` into `RFC6750AuthenticationFailureHandler`, which looks more appropriate here.